PSKoans/Public/Show-Karma.ps1

```diff
@@ -80,6 +85,30 @@ function Show-Karma {
                 $KoanLocation | Invoke-Item
             }
         }
+        'OpenFile' {
+            try {
+                $Results = Get-Karma @GetParams
+            }
+            catch {
+                $PSCmdlet.ThrowTerminatingError($_)
+            }
+            $UnformatedPath     = $Results.Meditation
+            $PSKoansFilePath = $UnformatedPath.Substring($UnformatedPath.IndexOf(', ')+2,$UnformatedPath.IndexOf(': ')-($UnformatedPath.IndexOf(', ')+2))
+            $PSKoansLine = $UnformatedPath.Substring($UnformatedPath.IndexOf(': line ')+7,$UnformatedPath.LastIndexOf(("`n"))-($UnformatedPath.IndexOf(': line ')+7))    
+            $FullArgument = "-g ${PSKoansFilePath}:${PSKoansLine}"
+            $Editor = Get-PSKoanSetting -Name Editor
+            if ($Editor -and (Get-Command -Name $Editor -ErrorAction SilentlyContinue)) {
+                $EditorSplat = @{
+                    FilePath     = $Editor
+                    ArgumentList = $FullArgument
+                    NoNewWindow  = $true
+                }
+                Start-Process @EditorSplat
+            }
+            else {
+                $PSKoanLocationFullPath | Invoke-Item
+            }
```

Comment:
Somewhere in here we need to check the editor in use, I think? Since these parameters are only valid for VS Code...

If it's not `code` or `code-insiders` then we need to just default to opening the file; we can pave the way forward with a `switch` statement on `$Editor`, which would allow us to easily extend it later with alternate editors and the parameters they support for this kind of thing.

If it _is_ VS Code I'd recommend we also add the `-r` switch so that it opens in the current window and they don't end up with a dozen VS Code windows open. 🙂 

Also, the `else` here should probably just directly open the koans file itself rather than the folder, since the user would have asked for the file itself to be opened.